### PR TITLE
GQSS-4585 Fixed generation of template.properties file for templates

### PR DESCRIPTION
**Pull request by nnesic**
created from snapshots on Nfs storage

### CODE REVIEW

File: services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java [hunk 1]
```diff
@@ -499,64 +499,60 @@ protected Answer copySnapshotToTemplateFromNfsToNfs(CopyCommand cmd, SnapshotObj
...
+                newTemplate.setPhysicalSize(prop.getPhysicalSize());
+                return new CopyCmdAnswer(newTemplate);
+            } catch (ConfigurationException e) {
+                s_logger.debug("Failed to create template:" + e.toString());
+                return new CopyCmdAnswer(e.toString());
+            } catch (InternalErrorException e) {
                 s_logger.debug("Failed to create template:" + e.toString());
                 return new CopyCmdAnswer(e.toString());
             }
+            catch (IOException e) {
```
Comment (HrWiggles) [inline services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java]: Minor nitpick: would have liked to see `catch` on same line as closing curly (if for nothing else but to be consistent with the two other `catch` statements above).

File: services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java [hunk 2]
```diff
@@ -499,64 +499,60 @@ protected Answer copySnapshotToTemplateFromNfsToNfs(CopyCommand cmd, SnapshotObj
...
+                s_logger.debug("Failed to create template:" + e.toString());
+                return new CopyCmdAnswer(e.toString());
+             }
```
Comment (HrWiggles) [inline services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java]: All three `catch` handlers contain the exact same code...
Maybe join them all into one?

Comment (nnesic) [inline services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java]: Yes.

File: services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java [hunk 3]
```diff
@@ -499,61 +499,50 @@ protected Answer copySnapshotToTemplateFromNfsToNfs(CopyCommand cmd, SnapshotObj
...
+                loc.create(1, true, destData.getName());
+                loc.addFormat(info);
+                loc.save();
+
+                try (// append physical size info
+                    FileWriter writer = new FileWriter(metaFile, true);
+                    BufferedWriter bufferWriter = new BufferedWriter(writer);
+                    ){
+                    long size = _storage.getSize(destFileFullPath);
+                    bufferWriter.write("size=" + size);
```
Comment (HrWiggles) [inline services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java]: Before, `uniquename` and `filename` were also written to the file before `size`.
Are `uniquename` and `filename` not needed?

Comment (nnesic) [inline services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java]: They are, but the TemplateLocator already takes care of writing them, so they would be written in duplicate.

**Comment by HrWiggles** [comment]
lgtm

**Pull request merged**